#!/usr/bin/env python2
PACKAGE = "vio_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("im_noise",          double_t, 0, "Image (tracker) nose",            0.5, 0, 10)
gen.add("acc_noise",         double_t, 0, "Accelerometer measurement noise", 0.5, 0, 10)
gen.add("gyro_noise",        double_t, 0, "Gyroscope measurement noise",     0.5, 0, 10)
gen.add("orientation_noise", double_t, 0, "Orientation measurement noise",   0.5, 0, 10)
gen.add("pressure_noise",    double_t, 0, "Pressure measurement noise",      0.5, 0, 10)
gen.add("sigma_init",        double_t, 0, "Inverse depth init uncertainty",  0.5, 0, 1)

gen.add("Kp_pos",            double_t, 0, "Proportional gain position",      0.5, 0, 10)
gen.add("Kd_pos",            double_t, 0, "Derivative gain position",        0.6, 0, 10)
gen.add("Kp_yaw",            double_t, 0, "Proportional gain yaw",           0.5, 0, 10)

exit(gen.generate(PACKAGE, "vio_ros", "vio_ros"))
